### ISSUE TITLE
stress: failed test in cockroach/sql/sql.test: TestPurgeOldLeases

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#6847]
Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/82e5c27735be951890f22fc85888a2cf46a13605

Stress build found a failed test:

```
=== RUN   TestPurgeOldLeases
W160522 07:03:16.003824 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160522 07:03:16.005940 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160522 07:03:16.007259 server/node.go:401  store store=0:0 ([]=) not bootstrapped
W160522 07:03:16.008908 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160522 07:03:16.009245 storage/replica_command.go:1470  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406639h3m17.008504747s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-22 07:03:16.009139725 +0000 UTC]
I160522 07:03:16.009736 server/node.go:327  **** cluster {4b294055-e1e8-42f3-95e4-88714a64f560} has been created
I160522 07:03:16.009758 server/node.go:328  **** add additional nodes by specifying --join=127.0.0.1:36326
W160522 07:03:16.009768 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160522 07:03:16.010496 server/node.go:414  initialized store store=1:1 ([]=): {Capacity:10365558784 Available:7970807808 RangeCount:0}
I160522 07:03:16.010543 server/node.go:302  node ID 1 initialized
I160522 07:03:16.010751 storage/stores.go:287  read 0 node addresses from persistent storage
I160522 07:03:16.010885 server/node.go:535  connecting to gossip network to verify cluster ID...
I160522 07:03:16.011401 server/node.go:556  node connected via gossip and verified as part of cluster {"4b294055-e1e8-42f3-95e4-88714a64f560"}
I160522 07:03:16.011449 server/node.go:355  [node=1] Started node with [[]=] engine(s) and attributes []
I160522 07:03:16.011490 server/server.go:416  starting https server at 127.0.0.1:34563
I160522 07:03:16.011518 server/server.go:417  starting grpc/postgres server at 127.0.0.1:36326
I160522 07:03:16.011926 storage/split_queue.go:100  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I160522 07:03:16.012868 storage/replica_command.go:2011  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160522 07:03:16.020825 server/updates.go:155  No previous updates check time.
I160522 07:03:16.055882 storage/replica_command.go:1470  range 2: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406639h3m17.109982998s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-22 07:03:16.055784696 +0000 UTC]
I160522 07:03:16.056019 storage/replica_command.go:2011  initiating a split of range=2 [/Table/11-/Max) at key /Table/12
I160522 07:03:16.068373 storage/replica_command.go:1470  range 3: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406639h3m17.118035764s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-22 07:03:16.068331008 +0000 UTC]
I160522 07:03:16.068487 storage/replica_command.go:2011  initiating a split of range=3 [/Table/12-/Max) at key /Table/13
I160522 07:03:16.111656 storage/replica_command.go:1470  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406639h3m17.161255623s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-22 07:03:16.111558814 +0000 UTC]
I160522 07:03:16.111880 storage/replica_command.go:2011  initiating a split of range=4 [/Table/13-/Max) at key /Table/14
I160522 07:03:16.265438 storage/replica_command.go:1470  range 5: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406639h3m17.315065417s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-22 07:03:16.265391987 +0000 UTC]
I160522 07:03:16.265570 storage/split_queue.go:100  splitting range=5 [/Table/14-/Max) at keys [/Table/50/0]
I160522 07:03:16.265955 storage/replica_command.go:2011  initiating a split of range=5 [/Table/14-/Max) at key /Table/50
I160522 07:03:16.295464 storage/replica_command.go:1470  range 6: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406639h3m17.34525492s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-22 07:03:16.295422912 +0000 UTC]
I160522 07:03:16.296413 storage/split_queue.go:100  splitting range=6 [/Table/50-/Max) at keys [/Table/51/0]
I160522 07:03:16.296721 storage/replica_command.go:2011  initiating a split of range=6 [/Table/50-/Max) at key /Table/51
--- FAIL: TestPurgeOldLeases (0.31s)
	lease_internal_test.go:153: sql/lease.go:206: DELETE FROM system.lease WHERE (descID, version, nodeID, expiration) = ($1, $2, $3, $4): expected 1 result, found 0
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
0 runs so far, 0 failures, over 25s
0 runs so far, 0 failures, over 30s
0 runs so far, 0 failures, over 35s
0 runs so far, 0 failures, over 40s
0 runs so far, 0 failures, over 45s
12 runs so far, 0 failures, over 50s
16 runs so far, 0 failures, over 55s
16 runs so far, 0 failures, over 1m0s
16 runs so far, 0 failures, over 1m5s
16 runs so far, 0 failures, over 1m10s
16 runs so far, 0 failures, over 1m15s
16 runs so far, 0 failures, over 1m20s
16 runs so far, 0 failures, over 1m25s
16 runs so far, 0 failures, over 1m30s
16 runs so far, 0 failures, over 1m35s
27 runs so far, 0 failures, over 1m40s
32 runs so far, 0 failures, over 1m45s
32 runs so far, 0 failures, over 1m50s
32 runs so far, 0 failures, over 1m55s
32 runs so far, 0 failures, over 2m0s
32 runs so far, 0 failures, over 2m5s
32 runs so far, 0 failures, over 2m10s
32 runs so far, 0 failures, over 2m15s
32 runs so far, 0 failures, over 2m20s
32 runs so far, 0 failures, over 2m25s
36 runs completed, 1 failures, over 2m26s
FAIL
```
Please assign, take a look and update the issue accordingly.

**Comment by tschottdorf** [comment]
@andreimatei assigned based on blame, but feel free to throw it at someone else.

**Comment by andreimatei** [comment]
This appears to be fixed by #6834, but this error is unexpected. Looking.
